### ISSUE TITLE
run test.sh  errors 

### ISSUE THREAD

**Issue by Hjy20255** [salathegroup/plantvillage-analysis#1]
i run test.sh
layer {
  name: "hdf5out"
  type: "HDF5Output"
  bottom: "prob"
  bottom: "label"
  hdf5_output_param {
    file_name: "hdf5_dumps/iter_13333.h5"
  }
  include {
    phase: TEST
  }
}

I1130 11:12:21.947036 19858 hdf5_output_layer.cpp:32] Saving HDF5 file hdf5_dumps/iter_13333.h5
I1130 11:12:21.947201 19858 hdf5_output_layer.cpp:37] Successfully saved 100 rows
I1130 11:12:21.947577 19858 caffe.cpp:285] Batch 0, accuracy = 1
I1130 11:12:21.947603 19858 caffe.cpp:285] Batch 0, loss = 0.00257298
I1130 11:12:22.067667 19858 hdf5_output_layer.cpp:32] Saving HDF5 file hdf5_dumps/iter_13333.h5
HDF5-DIAG: Error detected in HDF5 (1.8.11) thread 139774091323968:
  #000: ../../../src/H5D.c line 170 in H5Dcreate2(): unable to create dataset
    major: Dataset
    minor: Unable to initialize object
  #001: ../../../src/H5Dint.c line 439 in H5D__create_named(): unable to create and link to dataset
    major: Dataset
    minor: Unable to initialize object
  #002: ../../../src/H5L.c line 1638 in H5L_link_object(): unable to create new link to object
    major: Links
    minor: Unable to initialize object
  #003: ../../../src/H5L.c line 1882 in H5L_create_real(): can't insert link
    major: Symbol table
    minor: Unable to insert object
  #004: ../../../src/H5Gtraverse.c line 861 in H5G_traverse(): internal path traversal failed
    major: Symbol table
    minor: Object not found
  #005: ../../../src/H5Gtraverse.c line 641 in H5G_traverse_real(): traversal operator failed
    major: Symbol table
    minor: Callback failed
  #006: ../../../src/H5L.c line 1674 in H5L_link_cb(): name already exists
    major: Symbol table
    minor: Object already exists
F1130 11:12:22.067836 19858 hdf5.cpp:101] Check failed: status >= 0 (-1 vs. 0) Failed to make float dataset data
*** Check failure stack trace: ***
    @     0x7f1fb03ccdaa  (unknown)
    @     0x7f1fb03ccce4  (unknown)
    @     0x7f1fb03cc6e6  (unknown)
    @     0x7f1fb03cf687  (unknown)
    @     0x7f1fb0aab3cd  caffe::hdf5_save_nd_dataset<>()
    @     0x7f1fb09e3db2  caffe::HDF5OutputLayer<>::SaveBlobs()
    @     0x7f1fb0af7717  caffe::HDF5OutputLayer<>::Forward_gpu()
    @     0x7f1fb0a5de98  caffe::Net<>::ForwardFromTo()
    @     0x7f1fb0a5e217  caffe::Net<>::Forward()
    @           0x409e3b  test()
    @           0x408688  main
    @     0x7f1faeecff45  (unknown)
    @           0x408e8d  (unknown)
    @              (nil)  (unknown)
